### PR TITLE
feat(versioning:regex): support build number

### PR DESCRIPTION
**Pull request by viceice**
## Changes:

The regex versioning now supports an optional `build` match group, which is handled as 4th version part.

## Context:

This way we can easily support rolling docker releases like:
```Dockerfile
# Support bitnami rolling releases
FROM bitnami/postgresql:12.6.0-debian-10-r87

# Support linuxserver rolling releases
FROM linuxserver/ddclient:v3.9.1-ls62
```

Test repo: https://github.com/viceice-tests/docker-bitnami-versioning/pulls

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/versioning/loose/generic.ts [hunk 1]
```diff
@@ -133,7 +134,51 @@ export abstract class GenericVersioningApi<
       : null;
   }
 
-  protected abstract _compare(version: string, other: string): number;
+  protected _compare(version: string, other: string): number {
```
Comment (rarkins) [inline lib/versioning/loose/generic.ts:137]: Any potential that this causes other versionings to now change behavior?

File: lib/versioning/regex/index.ts [hunk 2]
```diff
@@ -98,10 +98,6 @@ export class RegExpVersioningApi extends GenericVersioningApi<RegExpVersion> {
     );
   }
 
-  isStable(version: string): boolean {
```
Comment (rarkins) [inline lib/versioning/regex/index.ts:101]: What's the implication removing this? Is it checked somewhere else instead?

Comment (viceice) [reply lib/versioning/regex/index.ts:101]: Yes, it's same implementation as in base class.

File: lib/versioning/loose/generic.ts (hunk 1 above)
Comment (viceice) [reply lib/versioning/loose/generic.ts:137]: no, only regex versioning is using this base class

**Review by rarkins — APPROVED** [review]

**Comment by viceice** [comment]
In a later pr i'll try to workaout a solution with arbitrary numbers with can be mapped like `1.2.3-b(?<b0>\d+)\.(?<b1>\d+).(?<b2>\d+)` or something like that

**Pull request merged**